lib/core/backend/backendImport.ts

```diff
@@ -0,0 +1,177 @@
+      }
+      // If some collections have already been defined, only their last mappings will be retained
+      this._application._import.mappings[index] =
+        Object.assign({}, this._application._import.mappings[index], mappings[index]);
+    }
+  }
+
+  /**
+   * Import profiles.
+   *
+   * @example
+   * {
+   *   profileA: { profile definition },
+   *   profileB: { profile definition },
+   *   ...
+   * }
+   *
+   * @param profiles Object containing profiles and their definition
+   */
+  profiles (profiles: JSONObject) : void {
+    if (this._application.started) {
+      throw runtimeError.get('already_started', 'import');
+    }
+    else if (! isPlainObject(profiles)) {
+      throw assertionError.get('invalid_type', 'profiles', 'object');
+    }
+
+    // If some profiles have already been defined, only their last definition will be retained
+    this._application._import.profiles =
+      Object.assign({}, this._application._import.profiles, profiles);
+  }
+
+  /**
+   * Import roles
+   *
+   * @example
+   * {
+   *   roleA: { role definition },
+   *   roleB: { role definition },
+   *   ...
+   * }
+   *
+   * @param roles Object containing roles and their definition
+   */
+  roles (roles: JSONObject) : void {
+    if (this._application.started) {
+      throw runtimeError.get('already_started', 'import');
+    }
+    else if (! isPlainObject(roles)) {
+      throw assertionError.get('invalid_type', 'roles', 'object');
+    }
+
+    // If some roles have already been defined, only their last definition will be retained
+    this._application._import.roles =
+      Object.assign({}, this._application._import.roles, roles);
+  }
+
+  /**
+   * Import user mappings.
+   *
+   * @example
+   * {
+   *   properties: {
+   *     fieldA: { type: 'keyword' },
+   *     fieldB: { type: 'integer' },
+   *     ...
+   *   }
+   * }
+   *
+   * @param mappings User properties
+   */
+  userMappings (mappings: JSONObject) : void {
+    if (this._application.started) {
+      throw runtimeError.get('already_started', 'import');
+    }
+    else if (! isPlainObject(mappings)) {
+      throw assertionError.get('invalid_type', 'mappings', 'object');
+    }
+
+    this._application._import.userMappings = mappings;
+  }
+
+  /**
+   * Import users.
+   *
+   * @example
+   * {
+   *   kuidA: { user content },
+   *   kuidB: { user content },
+   * }
+   *
+   * @param users Object containing users and their content
+   * @param options onExistingUsers: Default to `skip`. Strategy to adopt when trying to create an already existing user.
+   */
+  users (
+    users: JSONObject,
+    options: { onExistingUsers?: 'fail' | 'overwrite' | 'skip' } = {}
+  ) : void {
+    if (this._application.started) {
+      throw runtimeError.get('already_started', 'import');
+    }
+    else if (! isPlainObject(users)) {
+      throw assertionError.get('invalid_type', 'users', 'object');
+    }
+    else if (options.onExistingUsers) {
```

Comment:
Shoulnt we log a warning if we actually skip?

Reply:
IMHO I don't think so because it's a wanted behavior but it's up for debate :wink: 
The way I see this:

1. First time the dev start Kuzzle, an admin user is added with `skip` option
2. The dev stop Kuzzle for X motive
3. Then he restart Kuzzle and expects to have its admin user but doesn't care anymore of what import is written as long as it has worked in the first place
4. The server will be stopped and started multiple times and the admin won't be mentionned but will be there durably

However, by focusing on this option for this comment, I realized that `fail` option makes no sense so I will delete it. Also I think it's not clear that `onExistingUsers` is global and applies for all `import.users` (previous and future unless a more recent option has been given), so I will add a precision in the documentation.

Reply:
As a fervent supporter of explicit behavior, I add my +1 to skip warnings.

Reply:
Ok! The majority has spoken, I will add an explicit feedback for these actions